pkgs/pkgs-lib/formats.nix

```diff
@@ -274,6 +274,70 @@ rec {
 
   };
 
+  # Format for defining configuration of some PHP services, that use "include 'config.php';" approach.
+  php = {finalVariable ? null}: with lib; let
+    toPHP = value: with builtins;
+      if value == null then "null" else
+      if value == true then "true" else
+      if value == false then "false" else
+      if isInt value || isFloat value then toString value else
+      if isString value then string value else
+      if isAttrs value then attrs value else
+      if isList value then list value else
+      abort "formats.php: should never happen (value = ${value})";
+
+    escapePHP = escape [ "'" ];
+    string = value: "'${escapePHP value}'";
+
+    listContent = values: concatStringsSep ", " (map toPHP values);
+    list = values: "[" + (listContent values) + "]";
+
+    attrsContent = values: pipe values [
+      (mapAttrsToList (k: v: "${toPHP k} => ${toPHP v}"))
+      (concatStringsSep ", ")
```

Comment:
For easier debugging, it might be nice to format the output so that it  is not squashed into a single unreadable line.

Reply:
Which formatter should I use?

Reply:
php-cs-fixer should be fine. Or alternately, you could do a minimal formatting for legibility by hand by tracking indentation level of the printed value.

Reply:
PHP-CS-Fixer is the way to go indeed.

Reply:
Looks like I'm dumb, but I can't figure out how to make php-cs-fixer to actually format file.

php-cs-fixer fix doesn't work...

Reply:
That works here:

```
❯ php-cs-fixer fix --rules=@PSR12 .
PHP CS Fixer 3.51.0 Insomnia by Fabien Potencier, Dariusz Ruminski and contributors.
PHP runtime: 8.2.19
Loaded config default.
 77/77 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

   1) tests/unit/TypedIterableAggregateTest.php
   2) tests/unit/SortIterableAggregateTest.php
   3) tests/static-analysis/closure-iterator.php
   4) src/SortIterableAggregate.php
   5) src/RandomIterableAggregate.php
   6) src/ConcatIterableAggregate.php
   7) src/UnpackIterableAggregate.php
   8) src/FilterIterableAggregate.php
   9) src/NormalizeIterableAggregate.php
  10) src/ChunkIterableAggregate.php
  11) src/PackIterableAggregate.php
  12) src/InterruptableIterableIteratorAggregate.php
  13) src/MapIterableAggregate.php
  14) src/StringIteratorAggregate.php
  15) src/IterableIteratorAggregate.php
  16) src/RandomIntegerAggregate.php
  17) src/ReductionIterableAggregate.php
  18) src/MultipleIterableAggregate.php
  19) src/ReduceIterableAggregate.php
  20) src/LimitIterableAggregate.php
  21) stable-sort.php

Fixed 21 of 77 files in 0.387 seconds, 16.000 MB memory used
```

Reply:
@drupol: Maybe I'm doing something wrong?
```
$ cat default.php
<?php

return ['attrs' => ['foo' => null],
'float' => 3.141000,
'int' => 10, 'str' => 'foo', 'hell' => ['attrs' => ['foo' => null], 'float' => 3.141000, 'int' => 10, 'str' => 'foo']];

$ php-cs-fixer fix --rules=@PSR12 default.php
PHP CS Fixer 3.51.0 Insomnia by Fabien Potencier, Dariusz Ruminski and contributors.
PHP runtime: 8.3.6
Loaded config default.
 1/1 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%


Fixed 0 of 1 files in 0.004 seconds, 14.000 MB memory used

$ cat default.php
<?php

return ['attrs' => ['foo' => null],
'float' => 3.141000,
'int' => 10, 'str' => 'foo', 'hell' => ['attrs' => ['foo' => null], 'float' => 3.141000, 'int' => 10, 'str' => 'foo']];
```

Reply:
You need to pass a directory, not a file.

Reply:
Passing a file will be soon available, ref: https://twitter.com/_Codito_/status/1789049130867650665

Reply:
Indeed, I'm also unable to get it working now. WTH ?!?!

Reply:
I'm afraid to write custom indentation code because I'm not a PHP programmer. Is there any other formatters for PHP?

Reply:
no.

Reply:
ugh... i tried php_codesniffer but it does nothing too.. php is cursed.